src/Symfony/Component/Console/Logger/ConsoleLogger.php

```diff
@@ -105,15 +105,23 @@ public function log($level, $message, array $context = array())
      */
     private function interpolate($message, array $context)
     {
-        // build a replacement array with braces around the context keys
-        $replace = array();
+        if (false === strpos($message, '{')) {
+            return $message;
+        }
+
+        $replacements = array();
         foreach ($context as $key => $val) {
-            if (!is_array($val) && (!is_object($val) || method_exists($val, '__toString'))) {
-                $replace[sprintf('{%s}', $key)] = $val;
+            if (null === $val || is_scalar($val) || (\is_object($val) && method_exists($val, '__toString'))) {
+                $replacements["{{$key}}"] = $val;
+            } elseif ($val instanceof \DateTime || $val instanceof \DateTimeInterface) {
```

Comment:
note to mergers: first condition should be removed starting from 3.3

Reply:
@dunglas can you please rebase+retarget, and take this comment into account meanwhile?